ignite/metrics/nlp/bleu.py

```diff
@@ -122,20 +127,35 @@ def __init__(
         smooth: str = "no_smooth",
         output_transform: Callable = lambda x: x,
         device: Union[str, torch.device] = torch.device("cpu"),
+        average: str = "macro",
     ):
         if ngram <= 0:
             raise ValueError(f"ngram order must be greater than zero (got: {ngram})")
         self.ngrams_order = ngram
         self.weights = [1 / self.ngrams_order] * self.ngrams_order
         self.smoother = _Smoother(method=smooth)
-        super(Bleu, self).__init__(output_transform=output_transform, device=device)
 
-    def _corpus_bleu(
-        self, references: Sequence[Sequence[Sequence[Any]]], candidates: Sequence[Sequence[Any]],
-    ) -> float:
-        p_numerators: Counter = Counter()
-        p_denominators: Counter = Counter()
+        if average not in ["macro", "micro"]:
+            raise ValueError(f'Average must be either "macro" or "micro" (got: {average})')
+        self.average = average
+        self.p_numerators: Counter = Counter()
+        self.p_denominators: Counter = Counter()
+        self.hyp_length_sum = 0
+        self.ref_length_sum = 0
+
+        super(Bleu, self).__init__(output_transform=output_transform, device=device)
 
+    @staticmethod
```

Comment:
Let’s remove the static decorator and use self.ngram_order instead of use an extra argument. See my edit comment below